merlin/models/tf/models/benchmark.py

```diff
@@ -94,3 +99,125 @@ def NCFModel(
+    deep_block: Block
+        Block (structure) of deep model.
+    embedding_dims : Optional[dict]
+        Dimensions of the embeddings, in order to specify different dimensions for different
+        features, for features not specified by this dict, default dimension
+        (embedding_dim_default) would be used to set their embeddings.by default None, then
+        embedding_dim_default would be the fixed embedding dimension for all features
+    embedding_dim : Optional[int]
+        Default dimension of feateures not specified in embedding_dims
+    wide_schema : Optional[Schema]
+        The 'Schema' of input features for wide model, by default no features would be sent to
+        wide model, and the model would become a pure deep model, if specified, only features
+        in wide_schema would be sent to wide model
+    deep_schema : Optional[Schema]
+        The 'Schema' of input features for deep model, by default all features would be sent to
+        deep model. deep_schema and wide_schema could contain the same features
+    prediction_tasks: Optional[Union[PredictionTask, List[PredictionTask], ParallelPredictionBlock]
+        The prediction tasks to be used, by default this will be inferred from the Schema.
+    embedding_option_kwargs: Optional[dict]
+        Additional arguments to provide to `EmbeddingOptions` object for embeddings tables setting.
+        Defaults to {}
+
+    Returns
+    -------
+    Model
+
+    """
+
+    prediction_tasks = parse_prediction_tasks(schema, prediction_tasks)
+
+    if embedding_dims and embedding_dim_default:
+        embedding_options = EmbeddingOptions(
+            embedding_dims=embedding_dims,
+            embedding_dim_default=embedding_dim_default,
+            **embedding_option_kwargs,
+        )
+    elif embedding_dims:
+        embedding_options = EmbeddingOptions(
+            embedding_dims=embedding_dims, **embedding_option_kwargs
+        )
+    elif embedding_dim_default:
+        embedding_options = EmbeddingOptions(
+            embedding_dim_default=embedding_dim_default, **embedding_option_kwargs
+        )
+    else:
+        embedding_options = EmbeddingOptions(**embedding_option_kwargs)
+
+    if not wide_schema:
+        warnings.warn("If not specify wide_schema, NO feature would be sent to wide " "model")
+        wide_schema = None
+
+    if not deep_schema:
+        deep_schema = schema
+
+    if len(deep_schema) > 0:
+        if not deep_input_block:
+            deep_input_block = InputBlock(
+                deep_schema,
+                embedding_options=embedding_options,
+                **kwargs,
+            )
+        deep_body = deep_input_block.connect(
+            deep_block, MLPBlock(dimensions=[1], no_activation_last_layer=True)
+        )
+
+    if len(wide_schema) > 0:
+        if not wide_input_block:
+            wide_input_block = InputBlock(
+                wide_schema,
+                post=AsSparseFeatures(),
```

Comment:
No sure if using AsSparseFeatures is correct, here I want to transfer categorical features into sparse matrix, I refer from the source code of `class EmbeddingFeatures(TabularBlock)`, which uses this bock as pre block, but inside `AsSparseFeatures`, it calls `tf.RaggedTensor.from_row_lengths(values, row_lengths).to_sparse()`, so I am not sure if category feature column in dataset can be transformed by `tf.RaggedTensor.from_row_lengths(values, row_lengths).to_sparse()`.

Reply:
Got it, thank you Gabriel!